audio/out/ao_coreaudio.c

```diff
@@ -89,7 +89,7 @@ static OSStatus render_cb_lpcm(void *ctx, AudioUnitRenderActionFlags *aflags,
 
     int64_t end = mp_time_ns();
     end += p->hw_latency_ns + ca_get_latency(ts) + ca_frames_to_ns(ao, frames);
-    int samples = ao_read_data_nonblocking(ao, planes, frames, end);
+    int samples = ao_read_data(ao, planes, frames, end);
```

Comment:
I think we should not use `samples` returned from `ao_read_data`, which may be less than requested. The original implementation did not use it and always submit buffers of requested length to CoreAudio. CoreAudio *expect* the buffer length be equal to requested.